File: CS3315/Assignments/Week3Assignment/topTenHuckFinn.py
#!/usr/bin/env python3

import re

f = open("HuckFinn.txt", errors='ignore')
lines = f.readlines()
f.close()
data = {}
for line in lines:
    line = line.lower()
    words = re.findall("[a-z']+", line)
    for word in words:
        if (word in data):
            data[word] += 1
        else:
            data[word] = 1
top_ten = sorted(data, key=data.get, reverse=True)[:10]
for key in top_ten:
    print("  ", data[key], key)

File: CS3315/Assignments/Week3Assignment/topTenHuckFinn.pl
#!/usr/bin/env perl

open($file, '<', 'HuckFinn.txt');
@whole_file = <$file>;
close($file);
%data = ();
foreach $line (@whole_file) {
    $line = lc $line;
    foreach $word ($line =~ /[a-z']+/g) {
        if (exists($data{$word})) {
            $data{$word} = $data{$word} + 1;
        } else {
            $data{$word} = 1;
        }
    }
}
@top_ten = (reverse sort {$data{$a} <=> $data{$b}} keys %data)[0..9];
foreach $key (@top_ten) {
    print "   $data{$key} $key\n";
}
